nautilus_core/portfolio/src/portfolio.rs

```diff
@@ -76,94 +155,616 @@ impl Portfolio {
+            //         Some(account) => account,
+            //         None => {
+            //             log::error!(
+            //                 "Cannot update maintenance (position) margin: no account registered for {}",
+            //                 instrument_id.venue
+            //             );
+            //             initialized = false;
+            //             break;
+            //         }
+            //     };
+
+            //     if matches!(account, AccountAny::Margin(_)) {
+            //         continue;
+            //     }
+
+            //     let instrument = match borrowed_cache.instrument(&instrument_id) {
+            //         Some(instrument) => instrument,
+            //         None => {
+            //             log::error!(
+            //                 "Cannot update maintenance (position) margin: no instrument found for {}",
+            //                 instrument_id
+            //             );
+            //             initialized = false;
+            //             break;
+            //         }
+            //     };
+
+            //     (account, instrument)
+            // };
+
+            // TODO: later
+            // Update positions logic here
+        }
+
+        // let open_count = {
+        //     let borrowed_cache = self.cache.borrow();
+        //     borrowed_cache.positions_open(None, None, None, None).len()
+        // };
+
+        // log::info!(
+        //     "Initialized {} open position{}",
+        //     open_count,
+        //     if open_count == 1 { "" } else { "s" }
+        // );
+
+        // self.initialized = initialized;
+        todo!();
     }
 
+    // REMOVE THIS COMMENT after msgbus
+    // NEEDS: quote,
     pub fn update_quote_tick(&mut self, quote: &QuoteTick) {
-        todo!()
+        self.unrealized_pnls.remove(&quote.instrument_id);
+
+        if self.initialized || !self.pending_calcs.contains(&quote.instrument_id) {
+            return;
+        }
+
+        let borrowed_cache = self.cache.borrow();
+        let account = borrowed_cache.account_for_venue(&quote.instrument_id.venue);
+        if account.is_none() {
+            log::error!(
+                "Cannot update tick: no account registered for {}",
+                quote.instrument_id.venue
+            );
+            return;
+        }
+
+        let instrument = borrowed_cache.instrument(&quote.instrument_id);
+        if instrument.is_none() {
+            log::error!(
+                "Cannot update tick: no instrument found for {}",
+                quote.instrument_id
+            );
+            return;
+        }
+
+        let orders_open = borrowed_cache.orders_open(None, Some(&quote.instrument_id), None, None);
+
+        // todo
+        // Initialize initial (order) margin
+        // let result_init = self.accounts.
     }
 
     pub fn update_account(&mut self, event: &AccountState) {
```

Comment:
`update_account` only needs a reference to the cache. It does not need access to the full state of the Portfolio. We can avoid the inner outer pattern by moving a clone of the cache the update account closure. Just like we did message bus in RiskEngine.